skrub/_datetime_encoder.py

```diff
@@ -18,78 +22,456 @@
     "microsecond": "us",
     "nanosecond": "N",
 }
-TIME_LEVELS: list[str] = list(WORD_TO_ALIAS.keys())
-AcceptedTimeValues = Literal[
-    "year",
-    "month",
-    "day",
-    "hour",
-    "minute",
-    "second",
-    "microsecond",
-    "nanosecond",
-]
-
-
-class DatetimeEncoder(BaseEstimator, TransformerMixin):
-    """Transform each datetime column into several numeric columns \
-    for temporal features (e.g. "year", "month", "day"...).
-
-    Constant extracted features are dropped; for instance, if the year is
-    always the same in a feature, the extracted "year" column won't be added.
+TIME_LEVELS = list(WORD_TO_ALIAS)
+
+
+def _is_pandas_format_mixed_available():
+    pandas_version = pd.__version__
+    min_pandas_version = "2.0.0"
+    return parse_version(min_pandas_version) < parse_version(pandas_version)
+
+
+MIXED_FORMAT = "mixed" if _is_pandas_format_mixed_available() else None
+
+
+def to_datetime(
+    X,
+    errors="coerce",
+    **kwargs,
+):
+    """
+    Convert argument to datetime. Return the input if not datetime-parsable.
```

Comment:
maybe describe the main use case first eg "convert columns in X that store dates as strings to proper datetimes"